src/odemis/driver/hamamatsurx.py

```diff
@@ -496,7 +495,7 @@ def _getDataFromBuffer(self):
                 # TODO only request scaling table if corresponding MD not available for this time range
                 if self.parent._streakunit.streakMode.value:
                     # TODO some sync problem might be here if a different command is in queue
-                    # in between ImgRingBufferGet and ImgDataGet: check again!
+                    # in between ImgRingBufferGet and ImgDataGet: check again! This was solved by a lock right??
```

Comment:
Well, my understanding is that it cannot happen, because the only caller of `ImgRingBufferGet()` and `ImgDataGet()` is this very function. Otherwise, indeed, a lock around _every_ call to these two functions (and the reading back) would probably solve the issue.